src/type/result.ts

```diff
@@ -0,0 +1,74 @@
+export type Result<T, E extends string = string> =
```

Comment:
簡易的なドキュメントを書くのはどうでしょう。ちょっと書いてみました！

```suggestion
/**
 * Result型。新しいErrorクラスを作らなくてもエラーハンドリングできる。
 *
 * 成功時はsuccess関数を使ってSuccessResultを返し、
 * エラー時はfailure関数を使ってFailureResultを返せば良い。
 *
 * 例：成功時はstringを返し、エラー時はErrorCodeA・ErrorCodeBを返す場合
 *
 * // 関数定義
 * function generateString(): Result<string, "ErrorCodeA" | "ErrorCodeB"> {
 *   // 成功時
 *   return success("ok string");
 *
 *  // エラー時
 *   return failure("ErrorCodeA", new Error("error message A"));
 *   return failure("ErrorCodeB", new Error("error message B"));
 *   return failure(new Error("other error"));
 * }
 *
 * // 使う側
 * const result = generateString();
 * if (result.ok) {
 *  // 成功時の処理
 *   console.log(result.value); // -> "ok string"
 * }
 * else {
 *   // エラー時の処理
 *   console.log(result.code); // -> "ErrorCodeA" | "ErrorCodeB" | undefined
 *   console.log(result.error); // -> Error
 * }
 *
 * // getValueOrThrowを使えばエラー時にResultErrorをthrowし直す
 * const value = getValueOrThrow(result);
 * console.log(value); // -> "ok string"
 */

export type Result<T, E extends string = string> =
```